File: peernexus-frontend/src/services/connectionService.js
import apiClient from "./apiClient.js";

export const connectionService = {
  getConnections: (params) =>
    apiClient.get("/api/connections", { params }).then((r) => r.data.data),
  getIncomingRequests: (params) =>
    apiClient.get("/api/connections/requests/incoming", { params }).then((r) => r.data.data),
  getOutgoingRequests: (params) =>
    apiClient.get("/api/connections/requests/outgoing", { params }).then((r) => r.data.data),
  getMutualConnections: (userId) =>
    apiClient.get(`/api/connections/mutual/${userId}`).then((r) => r.data.data),
  sendRequest: (recipientId) =>
    apiClient.post("/api/connections/request", { recipientId }).then((r) => r.data.data),
  acceptRequest: (id) =>
    apiClient.post(`/api/connections/${id}/accept`).then((r) => r.data.data),
  rejectRequest: (id) =>
    apiClient.post(`/api/connections/${id}/reject`).then((r) => r.data.data),
  cancelRequest: (id) =>
    apiClient.post(`/api/connections/${id}/cancel`).then((r) => r.data.data),
  removeConnection: (id) =>
    apiClient.delete(`/api/connections/${id}`).then((r) => r.data),
};

File: peernexus-backend/src/main/java/com/peernexus/peernexus/connection/service/ConnectionService.java
package com.peernexus.peernexus.connection.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.peernexus.peernexus.connection.dto.ConnectionRequest;
import com.peernexus.peernexus.connection.dto.ConnectionResponse;
import com.peernexus.peernexus.connection.dto.ConnectionUserSummary;

public interface ConnectionService {

    ConnectionResponse sendRequest(ConnectionRequest request);

    ConnectionResponse acceptRequest(Long connectionId);

    ConnectionResponse rejectRequest(Long connectionId);

    ConnectionResponse cancelRequest(Long connectionId);

    void removeConnection(Long connectionId);

    Page<ConnectionResponse> getConnections(Pageable pageable);

    Page<ConnectionResponse> getIncomingRequests(Pageable pageable);

    Page<ConnectionResponse> getOutgoingRequests(Pageable pageable);

    List<ConnectionUserSummary> getMutualConnections(Long userId);
}
